cmd/livepeer/livepeer.go

```diff
@@ -331,7 +333,9 @@ func main() {
 		}
 
 		if n.NodeType == core.BroadcasterNode {
-			n.Sender = pm.NewSender(n.Eth)
+			// TODO: Initialize Sender with an implementation
+			// of RoundsManager that reads from a cache
+			n.Sender = pm.NewSender(n.Eth, n.Eth)
```

Comment:
Why are we passing in the Eth client twice as an argument to `pm.NewSender` ? 

I understand that it implements both `Signer` and `RoundsManager` interfaces so why can't we just pass it once? 

Is this a design flaw in our code or is this a design flaw in go? 

Because as I see it the only option to "solve" this would be to have an additional package that defines the `pm.Ticket` and `pm.SenderInfo` types to be then used in both the `pm` and `eth` packages. Which is less clean/more code than just passing the same argument twice. But then again not allowing cyclical imports is considered to be enforcing good layered design. Seems quite paradoxal to me right now.

Reply:
> I understand that it implements both Signer and RoundsManager interfaces so why can't we just pass it once?

`pm.NewSender` accepts the `pm.Signer` and `pm.RoundsManager` interfaces separately to indicate that each of these interfaces could be implemented by a different struct. In this case, `LivepeerEthClient` happens to implement both the `pm.Signer` and `pm.RoundsManager` interfaces, but this need not always be the case. In fact, as mentioned in the TODO comment, the `pm.RoundsManager` interface will likely be implemented by a different struct later on i.e. a struct that can return values cached from the RoundsManager contract instead of requesting the values with each method invocation.

> Is this a design flaw in our code or is this a design flaw in go?

I think this looks strange because I chose not to include the caching implementation of `pm.RoundsManager` in this PR (ideally we can implement #492 first instead of adding more one off subscription code). Once we add the caching implementation of `pm.RoundsManager`, we won't be passing `n.Eth` as a parameter twice.